### PR TITLE
test: Remove obsolete test scripts for minishift, kubeadm, and update test readme

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
- Simplify the test documentation on running the integration tests to clarify that any K8s cluster can be used, and only support minikube in the instructions.
- The test scripts to start minishift and kubeadm are no longer in use by Jenkins or anyone else.
    
**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by satoru-takeuchi — APPROVED** [review]

File: tests/README.md [hunk 1]
```diff
@@ -1,115 +1,67 @@
...
 
-- Using Kubeadm
-```
-tests/scripts/kubeadm.sh up
-export KUBECONFIG=~/admin.conf
-tests/scripts/helm.sh up
-```
-- Using minikube
-```
+Copy the local build into minikube:
```
Comment (subhamkrai) [inline tests/README.md:58]:
```suggestion
Start the minikube cluster and copy the local build into minikube:
```
or maybe some other words as `Copy the local build into minikube` inside `Start Kubernetes` heading seems incomplete.

**Review by subhamkrai — CHANGES_REQUESTED** [review]

File: tests/README.md [hunk 2]
```diff
@@ -1,115 +1,67 @@
...
-## Requirements
-1. Docker version => 1.2 && < 17.0 (for other alternatives, see below)
-1. Ubuntu 18 (the framework has only been tested on this version)
-1. kernel >= 4.15
-1. Kubernetes with kubectl configured
-1. Rook
+## Install Kubernetes
 
-## Instructions
+You can choose any Kubernetes install of your choice.  The test framework only depends on kubectl being configured.
```
Comment (leseb) [inline tests/README.md:9]:
```suggestion
You can choose any Kubernetes install of your choice. The test framework only depends on `kubectl` being configured.
```

**Review by leseb — CHANGES_REQUESTED** [review]
nit

File: tests/README.md [hunk 3]
```diff
@@ -1,115 +1,67 @@
...
-```
-tests/scripts/kubeadm.sh up
-export KUBECONFIG=~/admin.conf
-tests/scripts/helm.sh up
-```
-- Using minikube
-```
+Start the minikube cluster and copy the local build into minikube:
+
+```console
```
Comment (travisn) [inline tests/README.md:60]: @leseb As a follow-up PR, want to update this section (or perhaps add a new section) with your steps for more efficiently iterating?

Comment (leseb) [reply tests/README.md:60]: Yes will do!

File: tests/README.md [hunk 4]
```diff
@@ -1,115 +1,67 @@
...
-### Install Helm
 Use [helm.sh](/tests/scripts/helm.sh) to install Helm and set up Rook charts defined under `_output/charts` (generated by build):
 
 - To install and set up Helm charts for Rook run `tests/scripts/helm.sh up`.
 - To clean up `tests/scripts/helm.sh clean`.
 
-**NOTE:** `*kubeadm.sh`, `minikube.sh` and `helm.sh` scripts depend on some artifacts under the `_output/` directory generated during build time,
-these scripts should be run from project root. e.g., `tests/script/kubeadm.sh up`.
+**NOTE:** These helper scripts depend on some artifacts under the `_output/` directory generated during build time.
+These scripts should be run from the project root. e.g., `tests/script/miniube.sh up`.
```
Comment (leseb) [inline tests/README.md:44]:
```suggestion
These scripts should be run from the project root. e.g., `tests/script/minikube.sh up`.
```

**Review by leseb — CHANGES_REQUESTED** [review]
nit

**Comment by travisn** [comment]
@Mergifyio rebase

**Review by leseb — APPROVED** [review]

**Pull request merged**